File: java-to-js/src/method-overloading.ts
const logLevels: Record<LogLevel, LogLevelDetail> = {
  info: { label: "Info", code: 0, color: "#808080FF" },
  warn: { label: "Warning", code: 1, color: "#a68a0d" },
  error: { label: "Error", code: 2, color: "#f0524f" },
};

// const logMessage = "Default Message";

logMessage("Default message");
logMessage("Error message", 2);
logMessage("Warning message", "warn");

function logMessage(message: string): void;
function logMessage(message: string, level: LogLevel): void;
function logMessage(message: string, code: number): void;

function logMessage(
  message: string,
  secondArg: LogLevel | number = "info",
): void {
  const logLevel =
    typeof secondArg === "number"
      ? logLevelByCode(secondArg)
      : logLevels[secondArg];

  console.log(`%c[${logLevel.label}]: ${message}`, `color: ${logLevel.color}`);
}

function logLevelByCode(code: number) {
  return Object.values(logLevels)[code]!;
}

type LogLevel = "info" | "warn" | "error";
type LogLevelDetail = {
  label: string;
  code: number;
  color: string;
};

File: java-to-js/src/MethodOverloading.java
void main () {
  logMessage("Default message");
  logMessage("Error message", 2);
  logMessage("Warning message", LogLevel.warning);
  System.out.println(logMessage);
}

final String logMessage = "Default Message";

void logMessage(String message) {
  logMessage(message, LogLevel.info); 
}

void logMessage(String message, int code) {
  var level = LogLevel.values()[code];
  logMessage(message, level);
}

void logMessage(String message, LogLevel level) {
  System.out.println(STR."\{level.colour}\{level.label}: \{message}\{RESET}");
}

final static String RESET = "\u001B[0m";

enum LogLevel {
  info("Info", 0, "\u001B[37m"),
  warning("Warning", 1, "\u001B[33m"),
  error("Error", 2, "\u001B[31m");

  final String label;
  final int code;
  final String colour;

  LogLevel(String label, int code, String colour) {
    this.label = label;
    this.code = code;
    this.colour = colour;
  }
}